File: Year 2/Semester 4/Programming and Projection Enironments/Week 3/C#ThirdWeek/Saptamana3/repository/IRepository.cs
using Saptamana3.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Saptamana3.repository
{
    internal interface IRepository<ID, E> where E : Entity<ID>
    {
        IEnumerable<E> GetAll();
        E FindById(ID id);
        bool Add(E entity);
        E Delete(ID id);
        E Update(E entity);
    }
}

File: Year 2/Semester 4/Programming and Projection Enironments/Week 7/SeventhWeekServerClient/Persistance/src/main/java/SeventhWeekServerClient/persistance/repository/IRepository.java
package SeventhWeekServerClient.persistance.repository;

import SeventhWeekServerClient.domain.Entity;
public interface IRepository<ID, E extends Entity<ID>>  {
    /**
     *
     * @param id of type ID (template)
     * @return found entity or null
     */
    E findById(ID id);
    Iterable<E> getAll();
    boolean add(E entity);
    boolean delete(ID id);
    boolean update(E entity);
}
